jsm.go

```diff
@@ -701,30 +706,63 @@ func (js *js) StreamInfo(stream string, opts ...JSOpt) (*StreamInfo, error) {
 	if cancel != nil {
 		defer cancel()
 	}
+
+	var i int
+	var subjectMessagesMap map[string]uint64 = nil
 	var req []byte
+
+	siOpts := StreamInfoRequest{}
 	if o.streamInfoOpts != nil {
-		if req, err = json.Marshal(o.streamInfoOpts); err != nil {
+		siOpts.DeletedDetails = o.streamInfoOpts.DeletedDetails
+		siOpts.SubjectsFilter = o.streamInfoOpts.SubjectsFilter
+	}
+
+	for {
+		siOpts.Offset = i
```

Comment:
This is changing the behavior in that it makes all StreamInfo request being a non empty request with an Offset (since I don't think this is omitempty). I would set a boolean in the "if" condition above that at least do the offset setting and marshal ONLY if the o.streamInfoOpts is not nil, or maybe rewrite in a way that you deal with simple info where no subject details is asked first, then deal with the cases where subjects are returned.